manta-crypto/src/mac.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Message Authentication Codes
+
+use crate::{constraint::Native, hash::BinaryHashFunction, key::KeyDerivationFunction};
+use core::marker::PhantomData;
+
+/// Message Authentication Code
+pub trait MessageAuthenticationCode<COM = ()> {
+    /// Key Type
+    type Key: ?Sized;
+
+    /// Message Type
+    type Message: ?Sized;
+
+    /// Digest Type
```

Comment:
The name here seems to be different from the spec. E.g., `Digest` v.s. `Tag`. Should we make it consistent?